doc/rst/developer/bestPractices/TestSystem.rst

```diff
@@ -370,20 +374,50 @@ instance:
      --x=true # foo.true.good
      --x=false # foo.false.good
 
+
 will compare test output to ``foo.true.good`` for the first execution and
 ``foo.false.good`` for the second.
 
 Any line that is unlabeled will use the default ``.good`` for that test.
 Undefined behavior will occur when both the ``.compopts`` and ``.execopts``
 files specify a ``.good`` file in this way.
 
-Using precomp and prediff files
-+++++++++++++++++++++++++++++++
+If you want to use use default arguments for the test but specify a different
+``.good`` file, you can add a line in your compopts/execopts file as follows
+(note the space before the #):
+
+  .. code-block:: bash
+
+      # foo.execopts
+       # foo.true.good
+
+
+Using precomp, preexec and prediff files
+++++++++++++++++++++++++++++++++++++++++
+
+When creating a ``.precomp``, ``.preexec`` or ``.prediff`` file, the file
+must be an executable. You can turn your script into an executable by running:
+``chmod +x foo.precomp``. For specifying these files for entire directories,
+the files should be named ``PRECOMP``, ``PREXEC``, and ``PREDIFF``,
+respectively.
+
+If you wish to have a system wide ``.prediff`` file, you can use the
+``CHPL_SYSTEM_PREDIFF`` environment variable that takes a comma-separated
+list of prediffs to run before every test.
+
+Using PRETEST
++++++++++++++
 
-When creating a ``.precomp`` or ``.prediff`` file, the file must be an
-executable. You can turn your script into an executable by running:
-``chmod +x foo.precomp``. 
+``PRETEST`` allows you to run a script once before any test is run in a
+directory. This can be used to set up a test, for example, by generating
+``.good`` files, or create/build other programs that are used by the test.
+The file must be an executable. You can turn your script into an executable by
+running: ``chmod +x PRETEST``.
 
+Note that if running tests, in parallel, the ``PRETEST`` script will not be run
+for any subdirectories and must be either duplicated or have a symbolic link to
+the parent directory. You can add a symlink to a file in a parent directory by
+running:  ``ln -s ../PRETEST PRETEST``
```

Comment:
Its less an issue about running tests in parallel and more about recursively running test directories. Maybe just drop the first part? So it would be "Note that the ``PRETEST`` script will not be run for any subdirectories..."